tests/integration/aof-multi-part.tcl

```
@@ -1166,5 +1166,138 @@ tags {"external:skip"} {
             set d2 [r debug digest]
             assert {$d1 eq $d2}
         }
+
+        test "AOF will not open incr file until the first AOFRW success when AOF is dynamically enabled" {
+            # Clear all data and delete appenddir to get an empty redis
+            r flushall
+            r config set appendonly no
+            catch {exec rm -rf $aof_dirpath}
+            r debug loadaof
```

why do you do LOADAOF, and not just FLUSHALL? (or even just start an empty server).
again, i think that LOADAOF should fail when the AOF is missing.

---

do you do that because you're counting on a specific sequence number on the assertions below?
in that case i think it's best to start a fresh server. safer and doesn't rely on the side effect of LOADAOF

---

Agree.